src/helpers/articles/rateArticleHelper.js

```diff
@@ -0,0 +1,49 @@
+import db from '../../sequelize/models';
+
+const { Article } = db;
+
+/**
+ * @exports ArticRatelehelper
+ * @class Articlehelper
+ * @description Helps to generate token and passwor hashing
+ * */
+class ArticleRatelehelper {
+  /**
+       * Check the environment
+       * @function validateArticleRated
+       * @param  {integer} id - Check the rate
+       * @param  {integer} user - Check the user
+       * @return {string} Validate the rate
+       */
+  static async validateArticleRated(id, user) {
+    const findarticle = await Article.findOne({ where: { id } });
+    if (findarticle) {
+      const articleOwner = await Article.findAll({
+        where: { id, authorId: user }
+      });
+      if (articleOwner.length) {
+        return 'Sorry! You cannot rate your article';
+      }
+    }
+    if (findarticle == null) {
+      return 'This Article does not exist';
+    }
+    return true;
```

Comment:
I think your short-hand code is not working because of this method `validateArticle` which returns two different data types depending on whether the rate is valid or not. So when it's invalid, it returns a string type, but when it's valid it returns a boolean! I think it's a better approach to have a function return only one type of data. Like for instance in this case, instead of returning `true`, you should return a string value like `valid` since you're already returning a string on invalid rate